engines/sci/graphics/frameout.cpp

```diff
@@ -55,31 +57,37 @@
 
 namespace Sci {
 
-GfxFrameout::GfxFrameout(SegManager *segMan, ResourceManager *resMan, GfxCoordAdjuster *coordAdjuster, GfxScreen *screen, GfxPalette32 *palette, GfxTransitions32 *transitions) :
+GfxFrameout::GfxFrameout(SegManager *segMan, ResourceManager *resMan, GfxCoordAdjuster *coordAdjuster, GfxPalette32 *palette, GfxTransitions32 *transitions, GfxCursor32 *cursor) :
 	_isHiRes(ConfMan.getBool("enable_high_resolution_graphics")),
 	_palette(palette),
+	_cursor(cursor),
 	_resMan(resMan),
-	_screen(screen),
 	_segMan(segMan),
 	_transitions(transitions),
 	_benchmarkingFinished(false),
 	_throttleFrameOut(true),
 	_throttleState(0),
-	// TODO: Stop using _gfxScreen
-	_currentBuffer(screen->getDisplayWidth(), screen->getDisplayHeight(), nullptr),
 	_remapOccurred(false),
 	_frameNowVisible(false),
-	_screenRect(screen->getDisplayWidth(), screen->getDisplayHeight()),
 	_overdrawThreshold(0),
 	_palMorphIsOn(false) {
 
-	_currentBuffer.setPixels(calloc(1, screen->getDisplayWidth() * screen->getDisplayHeight()));
-
-	// QFG4 is the only SCI32 game that doesn't have a high-resolution toggle
+	// QFG4 is the only SCI32 game that doesn't have a high-resolution version
 	if (g_sci->getGameId() == GID_QFG4) {
 		_isHiRes = false;
 	}
 
+	if (g_sci->getGameId() == GID_PHANTASMAGORIA) {
+		_currentBuffer = Buffer(630, 450, nullptr);
+	} else if (_isHiRes) {
+		_currentBuffer = Buffer(640, 480, nullptr);
+	} else {
+		_currentBuffer = Buffer(320, 200, nullptr);
+	}
```

Comment:
GK1 Mac actually does not have the enable_high_resolution_graphics flag, which also makes sense when it's hires only. It wouldn't make sense to offer that option. It even seems that GK1 floppy also didn't offer hires graphics (I can't check that, I only own multiple CD versions) and only the DOS CD version did. So GK1 DOS floppy defaults to low-res and GK Mac floppy shouldn't. The original code made sense in that regard.

It's also a difference between the obvious guesswork code and code that for example sets resolutions and enforces such modes etc. At least clone2727 owned that version of the game and he wouldn't add such code, when the game is low-res only. I know that he made that version "work" back then even with the other guess-work code, so enforcing hires simply can't be bad.

The commit even says 24e983610dfb118c943d6a1867bf3ea5d81e18e4
"SCI: Fix GK1 Mac views, the game now starts"

At the very least you could add a comment saying "originally we enforced hires for GK1 mac here, but no one owns it atm, so we can't verify to make absolutely sure", but atm the code is simply removed and would be forgotten about and that's like throwing away a wiki page just for fun.

Anyway are you sure that the hashes were properly taken? I mean that's the same logic. Maybe those are also bad, who knows. No one can currently check GK1 Mac. Maybe they are incomplete? How do you know that they weren't based on guesswork and are perfectly fine, when you can not verify them?

I personally have to assume that both were correct. And that wasn't complicated code. It simply enforces hires for GK1 Mac. Hell, it could have been a simple option as well inside the detection tables instead, so we are basically discussing about a simple data flag and nothing more. I mean atm GK1 Mac got the options like GK1 DOS floppy, which means no hires option. How do you know that this data is accurate? You can't.